### PR TITLE
fixing DAS-259: limiting the results returned by spark

### PR DESCRIPTION
**Pull request by swsachith**

### CODE REVIEW

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/internal/SparkAnalyticsExecutor.java [hunk 1]
```diff
@@ -780,13 +781,19 @@ private String replaceTableNamesInQuery(int tenantId, String query) {
         }
     }
 
-    private static AnalyticsQueryResult toResult(DataFrame dataFrame)
+    private AnalyticsQueryResult toResult(DataFrame dataFrame)
             throws AnalyticsExecutionException {
-        return new AnalyticsQueryResult(dataFrame.schema().fieldNames(),
-                                        convertRowsToObjects(dataFrame.collect()));
+        int resultsLimit = this.sparkConf.getInt("carbon.spark.results.limit", -1);
+        if(resultsLimit != -1)
```
Comment (lafernando) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/internal/SparkAnalyticsExecutor.java]: Fix code formatting (indentation), and put curly braces for even single line statements.

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/internal/SparkAnalyticsExecutor.java [hunk 2]
```diff
@@ -780,13 +781,19 @@ private String replaceTableNamesInQuery(int tenantId, String query) {
...
+            return new AnalyticsQueryResult(dataFrame.schema().fieldNames(),
+                    convertRowsToObjects(dataFrame.limit(resultsLimit).collect()));
+        else
+            return new AnalyticsQueryResult(dataFrame.schema().fieldNames(),
+                    convertRowsToObjects(dataFrame.collect()));
     }
 
-    private static List<List<Object>> convertRowsToObjects(Row[] rows) {
+    private
```
Comment (lafernando) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/internal/SparkAnalyticsExecutor.java]: Why is this split into two lines?

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/util/AnalyticsConstants.java [hunk 3]
```diff
@@ -117,4 +117,6 @@ private AnalyticsConstants() {
     public static final String SPARK_UI_PORT = "spark.ui.port";
     public static final String SPARK_RECOVERY_MODE = "spark.deploy.recoveryMode";
     public static final String SPARK_RECOVERY_MODE_FACTORY = "spark.deploy.recoveryMode.factory";
+
```
Comment (chanikag) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/util/AnalyticsConstants.java]: Why this space ?

Comment (swsachith) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/util/AnalyticsConstants.java]: fixed with the PR : https://github.com/wso2/carbon-analytics/pull/137

**Pull request merged**